db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/OperatorToString.java

```diff
@@ -82,23 +122,190 @@ public static String toString(PredicateList predicateList) {
+        }
+        catch(StandardException e) {
+            return "Bad SQL Expression";
+        }
+    }
+
+    /**
+     * Return a spark SQL expression given a Derby SQL expression, with column
+     * references indicating column names in the source DataFrame.
+     */
+    public static String opToSparkString(ValueNode operand) throws StandardException {
+        String retval = "";
+
+        // Do not throw any errors encountered.  An error condition
+        // just means we can't generate a valid spark representation
+        // of the SQL expression to apply to a native spark Dataset,
+        // so should not be considered a fatal error.
+        try {
+            OperatorToString vars =
+                new OperatorToString(true,
+                                     new MutableInt(0));
+            retval = opToString2(operand, vars);
+
+        }
+        catch (StandardException e) {
+            if (e.getSQLState() != SQLState.LANG_DOES_NOT_IMPLEMENT)
+                throw e;
+        }
+        return retval;
+    }
+
+    private static void throwNotImplementedError() throws StandardException {
+        throw StandardException.newException(SQLState.LANG_DOES_NOT_IMPLEMENT);
+    }
+
+    // We don't support REAL (float), because the way spark
+    // evaluates expressions involving float causes accuracy errors
+    // that don't occur when splice does the evaluation.
+    private static boolean sparkSupportedType(int typeFormatId) {
+        return (typeFormatId == BOOLEAN_TYPE_ID  ||
+                typeFormatId == DATE_TYPE_ID     ||
+                typeFormatId == CHAR_TYPE_ID     ||
+                typeFormatId == VARCHAR_TYPE_ID  ||
+                typeFormatId == LONGVARCHAR_TYPE_ID  ||
+                typeFormatId == TINYINT_TYPE_ID  ||
+                typeFormatId == SMALLINT_TYPE_ID ||
+                typeFormatId == INT_TYPE_ID      ||
+                typeFormatId == LONGINT_TYPE_ID  ||
+                typeFormatId == DECIMAL_TYPE_ID  ||
+                typeFormatId == DOUBLE_TYPE_ID   ||
+                typeFormatId == TIMESTAMP_TYPE_ID);
+    }
+
+    private static boolean isNumericTypeFormatID(int typeFormatId) {
+        return (typeFormatId == TINYINT_TYPE_ID  ||
+                typeFormatId == SMALLINT_TYPE_ID ||
+                typeFormatId == INT_TYPE_ID      ||
+                typeFormatId == LONGINT_TYPE_ID  ||
+                typeFormatId == DECIMAL_TYPE_ID  ||
+                typeFormatId == DOUBLE_TYPE_ID   ||
+                typeFormatId == REAL_TYPE_ID);
+    }
+
+    private static boolean isOverflowSensitive(ValueNode operand) throws StandardException {
+        return (operand.getTypeId().getTypeFormatId() == LONGINT_TYPE_ID ||
+                operand.getTypeId().getTypeFormatId() == DECIMAL_TYPE_ID ||
+                operand.getTypeId().getTypeFormatId() == DOUBLE_TYPE_ID);
+    }
+    private static void checkOverflowHidingCases(BinaryArithmeticOperatorNode bao,
```

Comment:
Do we need to worry about hidden overflow for operator other than BinaryOperator? for example, Coalesce, ConditionalNode?